Port process-inherit-coding-system-flag to Rust

mgyucht opened the pull request:
# What changed?
I ported `process-inherit-coding-system-flag` to Rust. This addresses issue #644.

# How is it tested?
I ran the following code in Remacs:
```
(defvar myproc (start-process "test" nil "ls"))
(equal nil (process-inherit-coding-system-flag myproc)) ;; => t
(set-process-inherit-coding-system-flag myproc t)
(equal t (process-inherit-coding-system-flag myproc)) ;; => t
```

I can add this to a proper test in test/rust_src/src/process-tests.el if that's useful.

shaleh commented on rust_src/src/process.rs:
```diff
@@ -327,4 +328,16 @@ pub fn waiting_for_user_input_p() -> bool {
+/// Return the value of inherit-coding-system flag for PROCESS. If this flag is
+/// t, `buffer-file-coding-system` of the buffer associated with process will
+/// inherit the coding system used to decode the process output.
+#[lisp_fn]
```
Because you immediately call a `as_foo_or_error` function and do not use the LispObject again you should use the `foo` type instead of the base LispObject. In this case, LispProcessRef. For the return you can just specify `bool` and avoid the explicit conversion.

mgyucht replied:
OK, sounds good.

mgyucht commented:
Addressed your comments! In the future, I'll write the tests first and then work on the port itself. I'm curious how changing the type from LispObject to LispProcessRef still works. Does the lisp_fn macro take care of this somehow?

mgyucht commented:
Oh, nevermind, I saw the `impl From<LispObject> for LispProcessRef` in lisp.rs.

shaleh commented:
Yes, as you saw the macro simply calls `foo.into()` to convert from the LispObject into the more specific type and then does the same on the return value to convert it into a LispObject. This way the individual functions are clear on the types expected and we do not have constant 'as_foo` calls.

Be aware that `as_foo_or_error` is used when the LispFoo type is requested. This means the function will immediately fail if the typing is wrong. Be sure that the function being ported has this semantic before using the type. Sometimes an Option<LispFoo> is a better choice. In others, LispObject is ok and we just check the type and immediately return without erroring.

shaleh commented:
I have dropped my previous review. Once the new tests been fixed we can merge this.

shaleh commented:
@mgyucht checkout the "Closes" I added to the git commit message. Whenever your PR closes or references an Issue you should add it to the commit message. Not the first line but in the body.
